### PR TITLE
feat: support price range in `alipay`/`wechat` providers

### LINKED ISSUE #129: 匹配规则支持金额
场景：自如收款 > N 为房租，否则为水电

### PR DESCRIPTION
**Pull request by Triple-Z**
## Description

resolves #129 

## Motivation and Context

#129 

## Dependencies 

None

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How has this been tested?

```
make test
```

## Is this change properly documented?

Yes.

### CODE REVIEW

**Comment by Triple-Z** [comment]
This feature's 80% code is written by Cursor, and I am amazed by its tremendous performance.

File: pkg/analyser/wechat/wechat.go [hunk 1]
```diff
@@ -93,15 +93,21 @@ func (w Wechat) GetAccountsAndTags(o *ir.Order, cfg *config.Config, target, prov
...
 			}
 		}
 		if r.TimestampRange != nil {
 			match, err = util.SplitFindTimeStampInterval(*r.TimestampRange, o.PayTime, match)
 			if err != nil {
-				log.Fatalf(err.Error())
+				log.Fatalf("Error in SplitFindTimeStampInterval: %v", err)
 			}
 		}
+		if r.MinPrice != nil && o.Money < *r.MinPrice {
```
Comment (ramsayleung) [inline pkg/analyser/wechat/wechat.go:105]: what is the expected behavior if two min prices overlap? Let's say we mistakenly have the following rules :
```yaml
    - category: 日用百货
      minPrice: 10
      targetAccount: Expenses:Groceries
    - category: 日用百货
      minPrice: 11
      maxPrice: 99.99
      targetAccount: Expenses:Food:Drink
```

if the grocery item in the bill costs 12, it matches both rules, but we will the pick this rule `targetAccount: Expenses:Food:Drink`, not sure whether it's the expected behavior.

File: pkg/analyser/wechat/wechat.go [hunk 2]
```diff
@@ -93,15 +93,21 @@ func (w Wechat) GetAccountsAndTags(o *ir.Order, cfg *config.Config, target, prov
...
+			match = false
+		}
+		if r.MaxPrice != nil && o.Money > *r.MaxPrice {
+			match = false
+		}
```
Comment (ramsayleung) [inline pkg/analyser/wechat/wechat.go:110]: I think we should also check the `MinPrice` and `MaxPrice`, make sure that `minPrice` is less than the `maxPrice`.

**Review by ramsayleung — COMMENTED** [review]
Leave a few comments with questions, let me know whether the questions I mention are by design.

File: README.md [hunk 3]
```diff
@@ -427,6 +427,7 @@ alipay:
   > - `11:00-13:00`
   > - `11:00:00-13:00:00`
   > 24 小时制，起始时间和终止之间之间使用 `-` 分隔。
+- `minPrice`（最小金额）和 `maxPrice`（最大金额）的区间匹配。
```
[the remark this answers is not in the document]
Comment (Triple-Z) [reply README.md:430]: This is weird, and same apperence on my github:
image

But unchanged lines are the same messy code, so maybe this problem is related to GitHub itself.

File: pkg/analyser/wechat/wechat.go (hunk 1 above)
Comment (Triple-Z) [reply pkg/analyser/wechat/wechat.go:105]: This is the expected behavior, we have already described this behavior on README.

image

File: pkg/analyser/wechat/wechat.go (hunk 2 above)
Comment (Triple-Z) [reply pkg/analyser/wechat/wechat.go:110]: SGTM, if the `minPrice > maxPrice`, then this rule is invalid, ignore it.

**Review by ramsayleung — APPROVED** [review]

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (ramsayleung) [reply README.md:430]: I suspect the README.md is somehow saved as GBK encoding, but GitHub assume the diff is UTF-8: https://github.com/desktop/desktop/blob/c464fff0c4b7d679960ff2e3ddac0bdda2cc1348/app/src/lib/git/diff.ts#L358-L361

But I've checked the encoding of `README.md`, it's encoded as UTF-8:
```
> file -I README.md
README.md: text/plain; charset=utf-8
```

It's kind of odd.

**Review by ramsayleung — APPROVED** [review]
Look good to me, thanks for your great work.

Happy Mid-Autumn Festival :) (if you are in the UTC+8 timezone)

**Pull request merged**